build.cake

```diff
@@ -545,6 +548,62 @@ Task("PackageZip")
     Zip(ZIP_IMG, zipPath);
 });
 
+Task("InstallSigningTool")
+    .Does(() =>
+    {
+        var result = StartProcess("dotnet.exe", new ProcessSettings {  Arguments = "tool install SignClient -g" });
+    });
+
+Task("SignNuGet")
+    .IsDependentOn("InstallSigningTool")
+    .IsDependentOn("PackageNuGet")
+    .Does(() =>
+    {
+        // Get the secret.
+        var secret = EnvironmentVariable("SIGNING_SECRET");
+        if(string.IsNullOrWhiteSpace(secret)) {
+            throw new InvalidOperationException("Could not resolve signing secret.");
+        }
+        // Get the user.
+        var user = EnvironmentVariable("SIGNING_USER");
+        if(string.IsNullOrWhiteSpace(user)) {
+            throw new InvalidOperationException("Could not resolve signing user.");
+        }
+
+        var signClientPath = Context.Tools.Resolve("SignClient.exe") ?? Context.Tools.Resolve("SignClient") ?? throw new Exception("Failed to locate sign tool");
+
+        var settings = File("./signclient.json");
+        //var filter = File("./signclient.filter");
+
+        // Get the files to sign.
+        var files = GetFiles(string.Concat(PACKAGE_DIR, "*.nupkg"));
```

Comment:
Does SignClient also sign the .dll and .exe files within the nupkg?

Reply:
No it doesn't sign the assemblies inside the package. Each file is submitted to be signed, so it can be a bit slow with too many things being signed. We could submit key assemblies for signing in the future though.